### PR TITLE
added a catch method to onJoinQuizSubmit method (Homepage)

### LINKED ISSUE #95: Snackbar not showing on invalid quiz code (Homepage)
**Describe the bug**
No snackbar appears to warn of a invalid quiz code when the user inputs an invalid code and presses the 'join' button on the Homepage

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to https://quiz.coffee/
2. Type invalid code into the Quiz Code field
3. Click 'Join' button

**Expected behaviour**
Snackbar appears warning the user that the quiz code is invalid

**Actual behaviour**
No user feedback is given

### PR DESCRIPTION
**Pull request by timloudon**
Closes #95

### CODE REVIEW

**Review by richielewis — COMMENTED** [review]
Coud you rename setOpen and its corresponding state variable to something more descriptive e.g.  setSnackbarOpen  or setQuizJoinErrorOpen or something.

File: quizmaster-client-app/src/components/HomePage/HomePage.tsx [hunk 1]
```diff
@@ -85,13 +85,18 @@ function HomePage() {
...
+    Axios.get(`/api/quizzes/${quizCode}/name`)
+      .then((res) => {
+        if (!res.data) {
+          setOpen(true);
+        } else {
+          history.push(`/quiz/${quizCode}/${res.data.toUrlFormat()}`);
+        }
+      })
+      .catch((err) => {
+        console.log("error: ", err);
```
Comment (calumtomeny) [inline quizmaster-client-app/src/components/HomePage/HomePage.tsx:97]: I don't think we should commit logs to the console. We could also check for a 404 error using `err.response.status` e.g.
```
if (err.response.status === 404) {
      throw new Error(`${err.config.url} not found`);
}
```

**Review by calumtomeny — APPROVED** [review]
Looks good!

File: quizmaster-client-app/src/components/HomePage/HomePage.tsx (hunk 1 above)
Comment (richielewis) [reply quizmaster-client-app/src/components/HomePage/HomePage.tsx]: Is error 404 the code returned when the quiz code is not valid, or is error 404 a different error?

Comment (richielewis) [reply quizmaster-client-app/src/components/HomePage/HomePage.tsx]: Just checked and it is 404 ,   but what happens if its an error other than 404?  In this case the error will be "caught" and then not handled,  perhaps we need an ELSE here to throw the exception if its something other than 404

Comment (timloudon) [reply quizmaster-client-app/src/components/HomePage/HomePage.tsx]: Calum's mentioned that there might be other errors which would probably be handled globally (i.e. a server error even if the quiz code is correct etc.) and can be sorted latet

**Pull request merged**